test/python/test_identifiers.py

```diff
@@ -306,1107 +308,246 @@ def test_add_circuit_noname(self):
         self.assertRaises(QISKitError, result.get_counts)
 
 
-class TestZeroIds(QiskitTestCase):
-    """Circuits and records can have zero as names"""
+class TestQobj(QiskitTestCase):
+    """Check the objects compiled for different backends create names properly"""
 
     def setUp(self):
-        self.QPS_SPECS_ZEROS = {
-            "circuits": [{
-                "name": 0,
-                "quantum_registers": [{
-                    "name": 0,
-                    "size": 3}],
-                "classical_registers": [{
-                    "name": "",
-                    "size": 3}]
-            }]
-        }
+        qp = QuantumProgram()
+        qr = qp.create_quantum_register("qr2", 2)
+        cr = qp.create_classical_register(None, 2)
+        qc = qp.create_circuit("qc0", [qr], [cr])
+        qc.h(qr[0])
+        qc.measure(qr[0], cr[0])
+        qp.add_circuit("qc10", qc)
+        circuits = ["qc10"]
+        self.qr_name = qr.name
+        self.cr_name = cr.name
+        self.qp = qp
+        self.circuits = circuits
+
+    def test_local_qasm_simulator(self):
+        backend = 'local_qasm_simulator'
+        qobj = self.qp.compile(self.circuits, backend=backend, shots=1024)
+        cc = qobj['circuits'][0]['compiled_circuit']
+        ccq = qobj['circuits'][0]['compiled_circuit_qasm']
+        self.assertIn(self.qr_name, map(lambda x: x[0], cc['header']['qubit_labels']))
+        self.assertIn(self.qr_name, ccq)
+        self.assertIn(self.cr_name, map(lambda x: x[0], cc['header']['clbit_labels']))
+        self.assertIn(self.cr_name, ccq)
+
+    @unittest.skipIf('local_clifford_simulator' not in available_backends,
+                     'local_clifford_simulator is not available')
+    def test_local_clifford_simulator(self):
+        backend = 'local_clifford_simulator'
+        qobj = self.qp.compile(self.circuits, backend=backend, shots=1024)
+        cc = qobj['circuits'][0]['compiled_circuit']
+        ccq = qobj['circuits'][0]['compiled_circuit_qasm']
+        self.assertIn(self.qr_name, map(lambda x: x[0], cc['header']['qubit_labels']))
+        self.assertIn(self.qr_name, ccq)
+        self.assertIn(self.cr_name, map(lambda x: x[0], cc['header']['clbit_labels']))
+        self.assertIn(self.cr_name, ccq)
+
+    @unittest.skipIf('local_qiskit_simulator' not in available_backends,
+                     'local_qiskit_simulator is not available')
+    def test_local_qiskit_simulator(self):
+        backend = 'local_qiskit_simulator'
```

Comment:
Since #435 we now use `local_qasm_simulator_py` and `local_qasm_simulator_cpp` for the python vs. c++ simulator names. This part should be updated, otherwise it triggers a DeprecationWarning.

Reply:
Fixed by @ajavadia in https://github.com/QISKit/qiskit-sdk-py/pull/395/commits/a73b940a60c427c803b8219d780b5038ae0959e5